hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandomProvider.java

```diff
@@ -38,7 +40,8 @@ private ThreadLocalRandomProvider() {
     public static Random get() {
         Random random = THREAD_LOCAL_RANDOM.get();
         if (random == null) {
-            random = new Random();
+            long seed = SEED_GENERATOR.nextLong();
```

Comment:
Okay, so after some discussions and readings this is my actual state of knowledge:
* using the same `Random` instance from multiple threads should be fine since Java 5
* this might have an impact on performance, since there is a CAS loop inside of e.g. `Random::nextLong()`
* so for performance reasons we want to use multiple `Random` instances (here via `ThreadLocal`)
* there can still be an issue with generating multiple `Random` instances from multiple threads -> they can have same random seed
* this is the reason for the `SEED_GENERATOR`, which should be thread-safe due to the same `Random` instance being used